File: android-app/app/src/main/java/com/pranavakshit/gpscamportal/data/local/PhotoEntity.java
package com.pranavakshit.gpscamportal.data.local;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "photos")
public class PhotoEntity {
    @PrimaryKey(autoGenerate = true)
    public int id;
    public String locationName;
    public double latitude;
    public double longitude;
    public long timestamp;
    public String imageUri;
    public String uploader;
    public boolean isUploaded;
    public boolean isDeleted;
}

File: android-app/app/src/main/java/com/pranavakshit/gpscamportal/data/local/PhotoEntity.kt
package com.pranavakshit.gpscamportal.data.local

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "photos")
data class PhotoEntity(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    val locationName: String,
    val latitude: Double,
    val longitude: Double,
    val timestamp: Long,
    val imageUri: String, // Local URI path
    val uploader: String,
    val isUploaded: Boolean = false
)
